docs/tools/slurm.md

```diff
@@ -20,37 +20,116 @@ The following sections will provide detailed guidance on how to use SLURM to req
-    [using slurm on Grace-Hopper][gh200-slurm]
-    ```
-
-Link to the [Grace-Hopper overview][gh200-node].
-
-An example of using tabs to show srun and sbatch useage to get one GPU per MPI rank:
+The [GH200 nodes on Alps][gh200-node] have four GPUs per node, and SLURM job submissions must be configured appropriately to best make use of the resources.
+Applications that can saturate the GPUs with a single process per GPU should generally prefer this mode.
+[Configuring SLURM jobs to use a single GPU per rank][gh200-slurm-single-rank-per-gpu] is also the most straightforward setup.
+Some applications perform badly with a single rank per GPU, and require use of [NVIDIA's Multi-Process-Service (MPS)](https://docs.nvidia.com/deploy/mps/index.html) to oversubscribe GPUs with multiple ranks per GPU.
+
+The best SLURM configuration is application- and workload-specific, so it is worth testing which works best in your particular case.
+Also see [Scientific Applications][sciapps] for information about recommended application-specific SLURM configurations.
+
+!!! warning
+    The GH200 nodes have their GPUs configured in ["default" compute mode](https://docs.nvidia.com/deploy/mps/index.html#gpu-compute-modes).
+    Unlike "exclusive process" mode, "default" mode allows multiple processes to submit work to a single GPU simultaneously.
+    This also means that different ranks on the same node can inadvertently use the same GPU leading to suboptimal performance or unused GPUs, rather than job failures.
+    
+     Some applications benefit from using multiple ranks per GPU. However, [MPS should be used][gh200-slurm-multi-rank-per-gpu] in these cases.
+    
+     If you are unsure about which GPU is being used for a particular rank, print the `CUDA_VISIBLE_DEVICES` variable, along with e.g. `SLURM_LOCALID`, `SLURM_PROCID`, and `SLURM_NODEID` variables, in your job script.
+     If the variable is unset or empty all GPUs are visible to the rank and the rank will in most cases only use the first GPU. 
+
+[](){#gh200-slurm-single-rank-per-gpu}
+#### One rank per GPU
+
+Configuring SLURM to use one GH200 GPU per rank is easiest done using the `--ntasks-per-node=4` and `--gpus-per-task=1` SLURM flags.
+For advanced users, using `--gpus-per-task` is equivalent to setting `CUDA_VISIBLE_DEVICES` to `SLURM_LOCALID`, assuming the job is using four ranks per node.
+The examples below launch jobs on two nodes with four ranks per node using `sbatch` and `srun`:
 
 === "sbatch"
 
     ```bash
     #!/bin/bash
     #SBATCH --job-name=affinity-test
-    #SBATCH --ntasks-per-node=4
     #SBATCH --nodes=2
+    #SBATCH --ntasks-per-node=4
     #SBATCH --gpus-per-task=1
 
-    srun affinity
+    srun <application>
     ```
 
 === "srun"
 
     ```
-    > srun -n8 -N2 --gpus-per-task=1 affinity
+    srun --nodes=2 --ntasks-per-node=4 --gpus-per-task=1 <application>
     ```
+    
+Omitting the `--gpus-per-task` flag will lead to all ranks on the node using the first GPU.
```

Comment:
```suggestion
Omitting the `--gpus-per-task` flag results in `CUDA_VISIBLE_DEVICES` being unset, which will lead to all ranks on the node using the first GPU.
```
This is also something that changed at some point IIRC, where before all GPUs were visible with `CUDA_VISIBLE_DEVICES=0,1,2,3`?

Reply:
> This is also something that changed at some point IIRC, where before all GPUs were visible with CUDA_VISIBLE_DEVICES=0,1,2,3

I think my wording needs a bit of polishing... I don't think anything changed, it's just that most applications will use CUDA with the device set to 0 even when all devices are exposed. Let me try some clarification here.

Reply:
Sorry, probably my comment wasn't clear but it was related to https://github.com/eth-cscs/cscs-docs/pull/12#discussion_r1959327733, where I was suggesting it would be useful to have a sort of CHANGELOG. I think this changed at some point (not sure if from Todi to Daint or in a Daint updated), but it's un-related to what you wrote (where I just think clarifying that `CUDA_VISIBLE_DEVICES` will remain unset might be useful).

Reply:
Ah, I think I know what you're referring to now. The `--gpus-per-task` flag used to simply have no effect at all (or maybe result in jobs failing to start; in any case, it did not set `CUDA_VISIBLE_DEVICES`), but I'm pretty sure that was before users came on the system.

See other comment for the changelog (it exists on the current KB, but hasn't been ported yet).